### ISSUE TITLE
ADD LSP for mlir?

### ISSUE THREAD

**Issue by MATRIXKOO** [onnx/onnx-mlir#809]
When I play with MLIR, I found this [MLIR_LSP](https://mlir.llvm.org/docs/Tools/MLIRLSP/)
And I custom the language server.
I use it to check the .mlir results. Maybe I can add this to your repo? If the answer is yes, I will raise a PR.

**Comment by AlexandreEichenberger** [comment]
In general, we update MLIR in ONNX-MLIR from time to time. Last time we did it is with #761 . Are you suggesting that we update it again? If so, you are welcome to do it.

If this requires a different solution, again we are open to adding MLIR tools. It has to pass our CIs and be generally as non-disruptive as possible.

**Comment by MATRIXKOO** [comment]
Actually, I forked the office mlir language server. Because the Vscode Extension isn't impl whole feature. It's dialect-customed In that case, you can do 
![](
![image](https://user-images.githubusercontent.com/42641719/127667140-d6c084ee-30d4-49cb-9751-1f5b7ed26d94.png)
)
In vscode, I'd like to merge this with your project.

**Comment by AlexandreEichenberger** [comment]
Are you saying that you are interested in pushing changes that are not in MLIR main branch? If so, the best way to include it in ONNX-MLIR is to get it approved by MLIR first. We would be wary to complicate the merging of onnx-mlir with llvm-project repo.

I also use viscode and there is a mode that prettify .mlir file.

**Comment by AlexandreEichenberger** [comment]
@caoimhinuibrian you are a more advanced vscode user, maybe you can chime in.

**Comment by MATRIXKOO** [comment]
> 
> 
> Are you saying that you are interested in pushing changes that are not in MLIR main branch? If so, the best way to include it in ONNX-MLIR is to get it approved by MLIR first. We would be wary to complicate the merging of onnx-mlir with llvm-project repo.
> 
> I also use vscode and there is a mode that prettify .mlir file.

Yes, But it's not MLIR's main feature, just a stand-alone tool to help you to read mlir more comfortably. 
It's just a vscode extension. But need dialect sources to compile .

**Comment by caoimhinuibrian** [comment]
If I understand correctly, you want to contribute the LSP code for the onnx-mlir dialects? If so, please go ahead and submit a PR so we can look at what you are doing.

**Comment by MATRIXKOO** [comment]
Yes. You're right.I impl  mlir-lsp-server executable.
Vscode Extension is available on https://marketplace.visualstudio.com/items?itemName=MomenAbdelkarim-WyattCalandro-LuisPrieto.mlir
Install the Extension, and set The mlir-lsp-server executable path.
I will commit my code later.

**Comment by AlexandreEichenberger** [comment]
@MATRIXKOO let us know when your code will be uploaded.

**Comment by MATRIXKOO** [comment]
This week. I am a little busy recently.

**Comment by AlexandreEichenberger** [comment]
Closing this issue, established that we will consider the PR once the code is ready for review. Thanks for contributing.
